src/devices/Pn532/Pn532.cs

```diff
@@ -152,12 +154,17 @@ public Pn532(string portName, LogLevel logLevel = LogLevel.None)
         /// Create a PN532 using SPI
         /// </summary>
         /// <param name="spiDevice">The SPI Device</param>
+        /// <param name="pinChipSelect">The GPIO pin number for the chip select</param>
+        /// <param name="controller">A GPIO controller</param>
         /// <param name="logLevel">The log level</param>
-        public Pn532(SpiDevice spiDevice, LogLevel logLevel = LogLevel.None)
+        /// <param name="shouldDispose">Dispose the GPIO Controller at the end</param>
+        public Pn532(SpiDevice spiDevice, int pinChipSelect, GpioController? controller = null, LogLevel logLevel = LogLevel.None, bool shouldDispose = false)
         {
             LogLevel = logLevel;
             _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
-            _controller = new GpioController();
+            _shouldDispose = shouldDispose | controller == null;
+            _controller = controller ?? new GpioController();
+            _pin = pinChipSelect > 0 ? pinChipSelect : throw new ArgumentException($"Pin ChipSelect can only be positive");
```

Comment:
I'd recommend just check for "not -1", I assume for board numbering they might start numbering with 0, likely even logical numbering could start from 0

Reply:
good catch. I should actually check >=0 as anything negative is bad